crates/storage/db/src/abstraction/table.rs

```diff
@@ -53,9 +53,9 @@ pub trait Decode: Send + Sync + Sized + Debug {
 }
 
 /// Generic trait that enforces the database key to implement [`Encode`] and [`Decode`].
-pub trait Key: Encode + Decode + Ord + Clone {}
+pub trait Key: Encode + Decode + Ord + Clone + Serialize + for<'a> Deserialize<'a> {}
 
-impl<T> Key for T where T: Encode + Decode + Ord + Clone {}
+impl<T> Key for T where T: Encode + Decode + Ord + Clone + Serialize + for<'a> Deserialize<'a> {}
```

Comment:
hmm I think we can do this?

Reply:
I was forced to add serialize bounds because the `list` command can print `Key` as json, and `get` command deserializes a `Key` from json provided by user.